src/matcha_ml/state/remote_state_manager.py

```diff
@@ -274,11 +276,13 @@ def use_remote_state(self) -> Iterator[None]:
         Downloads the state before executing the code.
         Upload the state when context is finished.
         """
-        self.download(os.getcwd())
+        with Spinner("Downloading Matcha resources..."):
```

Comment:
I am not sure @JonoCX would approve of this i.e. using UI functions inside classes. But I don't have an elegant solution for this since this is a context manager.

Reply:
Yep, we can't add UI-based functionality into these classes - it should all exist in the CLI/UI layer.

What's the benefit of adding this? 

One solution would be for us add a generic spinner context manager in our UI layer but this is a separate ticket.